src/Metadata/Verifier/VerifierBase.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Tuf\Metadata\Verifier;
+
+use Tuf\Client\SignatureVerifier;
+use Tuf\Exception\FormatException;
+use Tuf\Exception\PotentialAttackException\FreezeAttackException;
+use Tuf\Exception\PotentialAttackException\RollbackAttackException;
+use Tuf\Metadata\MetadataBase;
+
+abstract class VerifierBase
+{
+
+    /**
+     * @var \Tuf\Metadata\MetadataBase
+     */
+    protected $trustedMetadata;
+
+    /**
+     * @var \Tuf\Client\SignatureVerifier
+     */
+    protected $signatureVerifier;
+
+    /**
+     * @var \DateTimeImmutable
+     */
+    protected $metadataExpiration;
+
+
+    /**
+     * MetaDataVerifierBase constructor.
+     *
+     * @param \Tuf\Client\SignatureVerifier $signatureVerifier
+     * @param \DateTimeImmutable $metadataExpiration
+     * @param \Tuf\Metadata\MetadataBase|null $trustedMetadata
+     */
+    public function __construct(SignatureVerifier $signatureVerifier, \DateTimeImmutable $metadataExpiration, ?MetadataBase $trustedMetadata = null)
+    {
+        $this->signatureVerifier = $signatureVerifier;
+        $this->metadataExpiration = $metadataExpiration;
+        if ($trustedMetadata && !$trustedMetadata->isTrusted()) {
```

Comment:
We should use `\Tuf\Metadata\MetadataBase::ensureIsTrusted` here now that it is public

Reply:
fixed